app/Controllers/Blog.php

```diff
@@ -0,0 +1,73 @@
+<?php namespace App\Controllers;
+
+use CodeIgniter\Exceptions\PageNotFoundException;
+use Config\Blog as BlogConfig;
+use App\Libraries\Blog as BlogLibrary;
+
+class Blog extends BaseController
+{
+    /**
+     * @var BlogConfig
+     */
+    protected $config;
+
+    /**
+     * @var BlogLibrary
+     */
+    protected $blog;
+
+    public function __construct()
+    {
+        $this->config = config(BlogConfig::class);
+        $this->blog = new BlogLibrary();
```

Comment:
```suggestion
        $this->blog = new BlogLibrary($this->config);
```
Injecting our instance (see below) should pass it by reference and make it easier to track.